Xsiam new validations (#2501)
demisto/demisto-sdk: CHANGELOG.md

## Unreleased
* Added the ability to limit the number of CPU cores with `DEMISTO_SDK_MAX_CPU_CORES` envirment variable.
* Fixed an issue where **update-release-notes** failed when changing only xif file in **Modeling Rules**.

> * Added a validation to validate types of attributes in the schema file of modeling rule.
> * Added a validation to validate that the dataset name is the same in the xif and schema files of modeling rule.
> * Added a validation to validate that a correlation rule file does not start with a hyphen.
> * Added a validation to validate that xsiam content items follow naming conventions.

Reviewer: ```suggestion
* Added a **validate** step checking types of attributes in the schema file of modeling rule.
* Added a **validate** step checking that the dataset name of a modeling rule shows in the xif and schema files.
* Added a **validate** step checking that a correlation rule file does not start with a hyphen.
* Added a **validate** step checking that xsiam content items follow naming conventions.
```
